File: go/wam/2496.go
const WamReceiptStanzaReceive = 2496 // channel: regular

type ReceiptStanzaReceiveEvent struct {
	ReceiptStanzaDuration string `wam:"receipt_stanza_duration"` // field 1
	MediaType MEDIATYPE `wam:"media_type"` // field 2
	ReceiptStanzaOfflineCount int64 `wam:"receipt_stanza_offline_count"` // field 3
	ReceiptStanzaType string `wam:"receipt_stanza_type"` // field 4
	ReceiptStanzaRetryVer int64 `wam:"receipt_stanza_retry_ver"` // field 5
	ReceiptStanzaHasOrphaned bool `wam:"receipt_stanza_has_orphaned"` // field 6
	ReceiptStanzaTotalCount int64 `wam:"receipt_stanza_total_count"` // field 7
	ReceiptStanzaProcessedCount int64 `wam:"receipt_stanza_processed_count"` // field 8
	ReceiptStanzaStage RECEIPTSTANZASTAGE `wam:"receipt_stanza_stage"` // field 9
	MessageType MESSAGETYPE `wam:"message_type"` // field 10
	ReceiptAggregation RECEIPTAGGREGATIONTYPE `wam:"receipt_aggregation"` // field 11
	DbReadsCount int64 `wam:"db_reads_count"` // field 12
	DbWritesCount int64 `wam:"db_writes_count"` // field 13
	ProcessingDeferred bool `wam:"processing_deferred"` // field 14
	StanzaBatchSize int64 `wam:"stanza_batch_size"` // field 15
	StanzaProcessCount int64 `wam:"stanza_process_count"` // field 16
}

File: rust/wam/2496.rs
pub const WAM_RECEIPT_STANZA_RECEIVE: u32 = 2496;

#[derive(Debug, Default)]
pub struct ReceiptStanzaReceiveEvent {
    /// field 1, wire `receipt_stanza_duration`
    pub receipt_stanza_duration: Option<String>,
    /// field 2, wire `media_type`
    pub media_type: Option<MEDIATYPE>,
    /// field 3, wire `receipt_stanza_offline_count`
    pub receipt_stanza_offline_count: Option<i64>,
    /// field 4, wire `receipt_stanza_type`
    pub receipt_stanza_type: Option<String>,
    /// field 5, wire `receipt_stanza_retry_ver`
    pub receipt_stanza_retry_ver: Option<i64>,
    /// field 6, wire `receipt_stanza_has_orphaned`
    pub receipt_stanza_has_orphaned: Option<bool>,
    /// field 7, wire `receipt_stanza_total_count`
    pub receipt_stanza_total_count: Option<i64>,
    /// field 8, wire `receipt_stanza_processed_count`
    pub receipt_stanza_processed_count: Option<i64>,
    /// field 9, wire `receipt_stanza_stage`
    pub receipt_stanza_stage: Option<RECEIPTSTANZASTAGE>,
    /// field 10, wire `message_type`
    pub message_type: Option<MESSAGETYPE>,
    /// field 11, wire `receipt_aggregation`
    pub receipt_aggregation: Option<RECEIPTAGGREGATIONTYPE>,
    /// field 12, wire `db_reads_count`
    pub db_reads_count: Option<i64>,
    /// field 13, wire `db_writes_count`
    pub db_writes_count: Option<i64>,
    /// field 14, wire `processing_deferred`
    pub processing_deferred: Option<bool>,
    /// field 15, wire `stanza_batch_size`
    pub stanza_batch_size: Option<i64>,
    /// field 16, wire `stanza_process_count`
    pub stanza_process_count: Option<i64>,
}